### DISCUSSION TITLE
Start hint: `WARNING: Could not find the Qt platform plugin "wayland" in ""`

### DISCUSSION THREAD

**Discussion by glyh** [qutebrowser/qutebrowser#7692] [Q&A] [answered]
**Version info**:

```
qutebrowser v2.5.4
Git commit: 346a39ba1 on master (2023-04-24 13:12:50 +0200)
Backend: QtWebEngine 5.15.13, based on Chromium 87.0.4280.144
Qt: 5.15.9 (compiled 5.15.8)
```

**Does the bug happen if you start with `--temp-basedir`?**:

Yes.

**Description**

Qutebrowser is using XWayland instead of wayland native on sway. But I have `qt6-wayland` installed. 

(This can be tell if you are using a HiDPI screen because it's a known issue that XWayland doesn't scale well on HiDPI screens)

Here's the log:
```
11:31:43 WARNING: Could not find the Qt platform plugin "wayland" in ""
11:31:44 INFO: Run :adblock-update to get adblock lists.

```

**How to reproduce**

1. Install sway on arch linux along with qt6-wayland(and other dependencies)
2. start qutebrowser

**Comment by The-Compiler** [comment]
Converted this to a discussion because this is a problem with your setup, not with qutebrowser - any other Qt application will show the same issue.

Please try `export QT_DEBUG_PLUGINS=1` and run it again, you should get some more output from Qt to diagnose this.

**Comment by bitraid** [comment, marked as answer]
You are using Qt5 - to use Qt6, set `QUTE_QT_WRAPPER=PyQt6` in your environment.

**Reply by glyh** [reply]
Yeah I also have to install `pyqt6`
